Dispatcher timer operations should run in dispatcher thread

WallyCZ opened the pull request:
#Fixes:
Possible fix for #884 

#Changes:

Dispatcher timer operations (as well as other things) should run in dispatcher thread, but DoLogout is called also from ExceptionHandler, what could be another thread

advancedrei commented:
Threading seems to have affected stability significantly. I've got all kinds of crashes in the currently checked-in build, but I was running (mostly) fine on last night's code. Might be time to freeze the codebase for a day or so and work out dispatching and threading app-wide.

WallyCZ commented:
Yes, there will be more places like this probably...

WallyCZ commented:
Updated: after small discussion with ST-Apps previous patch was useless, so I changed it to method, that really should be protected. The point is that we never know from which thread that code could be called and things using dispatcher must run under it.

ST-Apps commented:
Merging.
By the way, @advancedrei we have a Slack team and I'd like you to join so that we can coordinate before doing things. If you're in, I need an email to send the invitation.

advancedrei commented:
It's robert@advancedrei.com.
